### PR TITLE
Add analytics namespace

### LINKED ISSUE #35700: Can see move and archive buttons for metabase analytics collection in the root collection
https://metaboat.slack.com/archives/C01LQQ2UW03/p1699525120236779

### Describe the bug

Metabase analytics is supposed to be a read-only collection. So we shouldn't allow users to move or archive it, although these actions do nothing, it's confusing that we can even see them.

image

After moving or archive it shows this error, but after refreshing the page, the collection is still there
image

### To Reproduce

1. Go to the root collection
2. Select **Metabase analytics** collection
3. See that you can either **Move** or **Archive** it

### Expected behavior

I shouldn't be able to select this collection with bulk actions in the first place since it's a read-only collection

### Logs

_No response_

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.14.1+1-LTS",
    "java.vendor": "Azul Systems, Inc.",
    "java.vendor.url": "http://www.azul.com/",
    "java.version": "11.0.14.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.14.1+1-LTS",
    "os.name": "Mac OS X",
    "os.version": "14.0",
    "user.language": "en",
    "user.timezone": "Asia/Bangkok"
  },
  "metabase-info": {
    "databases": [
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.214 (2022-06-13)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.214 (2022-06-13)"
      }
    },
    "run-mode": "dev",
    "version": {
      "date": "2023-11-12",
      "src_hash": "93a2193e3bb61e49153bcf2aa91b6ee2110b129c",
      "tag": "v1.48.0-SNAPSHOT",
      "hash": "d00c3f6"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

Confusing, because these actions does nothing.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by qwef**
Closes https://github.com/metabase/metabase/issues/35700

### Description

Add analytics namespace and have filtering centered around the namespace instead of the previous way, which was more ad-hoc.

### CODE REVIEW

File: enterprise/backend/test/metabase_enterprise/audit_app/api/collection_test.clj [hunk 1]
```diff
@@ -15,12 +22,12 @@
         (testing "Instance Analytics Collection should be the last collection."
           (testing "GET /api/collection"
             (is (= "instance-analytics"
-                   (->> (mt/user-http-request :rasta :get 200 "collection")
+                   (->> (mt/user-http-request :crowberto :get 200 "collection")
```
Comment (qwef) [inline enterprise/backend/test/metabase_enterprise/audit_app/api/collection_test.clj:25]: Swapping to admin user as non admin default shouldn't be able to see metabase analytics

File: enterprise/backend/test/metabase_enterprise/audit_app/api/collection_test.clj [hunk 2]
```diff
@@ -29,11 +36,67 @@
...
+                   (= "instance-analytics" (:type tree)) (let [ids (transient #{})]
+                                                           (walk/postwalk
+                                                            (fn [x] (when (and (map? x) (:id x)) (conj! ids (:id x))) x)
+                                                            tree)
+                                                           (recur coll-tree (into ia-ids (persistent! ids))))
+                   ;; TODO: put my children onto the end of the coll-tree, then recur like normal
+                   :else (recur (concat coll-tree (:children tree)) ia-ids)))
+           (mapv (comp :name id->coll))))))
+
+(deftest list-collections-visible-collections-test
```
Comment (qwef) [inline enterprise/backend/test/metabase_enterprise/audit_app/api/collection_test.clj:68]: Moved this test to enterprise namespace, and also added `with-audit-db-restoration`.

File: src/metabase/api/collection.clj [hunk 3]
```diff
@@ -61,13 +61,6 @@
       (remove personal? collections)
       collections)))
 
-(defn- filter-audit-collection
```
Comment (qwef) [inline src/metabase/api/collection.clj:64]: Removed the old ad-hoc filtering

**Review by noahmoss — APPROVED** [review]
Awesome, looks great. I want to do a bit more bug bashing later just in case there are any more edge cases we forgot, but the code looks good so don't want to block you on that

**Pull request merged**